src/App.js

```diff
@@ -0,0 +1,54 @@
+import { checkRouteValidation } from './utils/Validation.js';
+import { init, routeChange } from './utils/Route.js';
+import { getRootAPI, getContentAPI } from './utils/API.js';
+import ListPage from './page/ListPage.js';
+import EditPage from './page/EditPage.js';
+
+export default class App {
+  constructor(target, rootDocument) {
+    this.$target = target;
+    this.$listPage = new ListPage(
+      this.$target,
+      rootDocument,
+      this.selectDocument
+    );
+    this.$editPage = null;
+    init(this.route);
+    this.route();
+  }
+
+  route = () => {
+    const { pathname } = location;
+
+    if (checkRouteValidation(pathname) === true) {
+      this.$editPage = new EditPage(
+        this.$target,
+        pathname,
+        this.selectDocument,
+        this.reflectTitleChange
+      );
+    }
+    else {
+      const $editPage = document.querySelector('.content-page-container');
+      this.$target.removeChild($editPage);
+    }
+    this.addRouteEvent();
+  };
+
+  addRouteEvent = () => {
+    window.addEventListener('popstate', this.route);
+  };
+
+  selectDocument = (selectedDocumentId) => {
```

Comment:
```suggestion
  selectDocument = (selectedDocumentId = null) => {
```
`JS`에서는 `undefined`를 주의해야되기때문에 `null`만 체크하는 것도 위험합니다 ㅎㅎ